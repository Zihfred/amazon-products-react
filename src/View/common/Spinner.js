import React from "react";
import spinnerImg from "./img/spinner.gif";
import PropTypes from "prop-types";

const Spinner = ({ isLoading }) => {
  return isLoading ? (
    <div className="spinnerWrapper">
      <img src={spinnerImg} alt="" />
    </div>
  ) : null;
};

Spinner.propTypes = {
  isLoading: PropTypes.bool
}
export default Spinner;
