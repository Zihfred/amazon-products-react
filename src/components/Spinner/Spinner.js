import React from 'react';
import spinnerImg from "../../img/spinner.gif";

const Spinner= ({isLoading}) => {
  if (isLoading) return (
    <div className="spinnerWrapper">
      <img src={spinnerImg} alt="" />
    </div>
  );

  return null;
};

export default Spinner;
