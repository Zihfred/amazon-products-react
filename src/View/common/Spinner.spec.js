import Spinner from "./Spinner";
import React from "react";
import { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { configure } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
configure({ adapter: new Adapter() });

describe("", () => {
  it("isLoading", function() {
    const wrapper = shallow(<Spinner isLoading={true} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
