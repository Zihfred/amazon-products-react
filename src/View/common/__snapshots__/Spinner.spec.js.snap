// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isLoading 1`] = `
<div
  className="spinnerWrapper"
>
  <img
    alt=""
    src="spinner.gif"
  />
</div>
`;
